lib/conrad/collector.rb

```diff
@@ -138,10 +138,12 @@ def add_event(event)
     #   emitted events, the error will be allowed to bubble up. This is to
     #   prevent the unexpected loss of events if a single one is malformed.
     def record_events
-      if emit_as_batch?
-        record_events_as_batch
-      else
-        record_individual_events
+      Array(emitter).each do |emitter|
+        if emit_as_batch?
+          record_events_as_batch(emitter.clone, events.clone)
```

Comment:
Curious: why are we cloning the emitter here?

Reply:
the thinking was that if we have a thread that enqueues and then exits i'm not sure what happens to the thread locals. it may not be necessary

Reply:
Hm. 
So thread 1 pushes things onto the queue
Thread 2 has the block that will be using that clojure to emit them?

I'm about 95% sure that the RubyVM can't garbage collect those items. The `events` makes sense (somewhat) to clone since it'll get cleared in here which _would_ wipe out the memory. The emitter should still exist for...a variety of reasons, so cloning shouldn't be necessary for the emitter.